### ISSUE TITLE
v3.3.0 – Content Relationship Field returns null

### ISSUE THREAD

**Issue by kieranstartup** [angeloashmore/gatsby-source-prismic#297]
Hey,

I've just set up a new project, installed v3.3.0 (without realising) and I think I've discovered a bug. When querying a content relationship field, the data returned inside the document is `null`. Here's an example of my query:

````
{
  prismicHomepage {
    data {
      projects {
        project {
          document {
            ... on PrismicProject {
              id
              data {
                title {
                  html
                }
              }
            }
          }
        }
      }
    }
  }
}
````

I just tried rolling back to v3.2.1 and it returns the data correctly.

Let me know if I can supply any other info to help you debug.

Thanks!

**Comment by angeloashmore** [comment]
Thanks @kieranstartup, I just came across it myself as well. Working on a fix. 👍

**Comment by angeloashmore** [comment]
Fix published in `v3.3.1`

@MarcMcIntosh Just want to confirm that the change made here ([link](https://github.com/angeloashmore/gatsby-source-prismic/commit/2badd85936511c21d61761da2d9b1daece40af94#diff-5d1b7aa4dedd927c74804cfae278a945918146b9412c21904274e8e78af7073dR208)) isn't a required change for the latest Gatsby Cloud changes that was just merged? The value passed to `createNode` on that line needs to be the same as here in order to get the same ID when linking the field to the linked document:

https://github.com/angeloashmore/gatsby-source-prismic/blob/4fb49edd94a1b9703c6178128f27c9db84b7f965/src/environment.node.ts#L110

**Comment by MarcMcIntosh** [comment]
Hi @angeloashmore 
Yes the Id's need to be the same, 
(I must have missed that line during when writing the webhhooks part). 
One of the changes that I made when writing the integration for was to use only the field.id for creating node ids.
This was because when a doc has been deleted we can not find out the type as only the id is sent in the webhook payload.

I'll have to have a look and see where else crearteNodeId is called.

**Comment by angeloashmore** [comment]
The latest fixes have been published in `v3.3.2`.

All link fields and the Gatsby Preview integration should work as expected now. Thanks everyone! 🎉
